### DISCUSSION TITLE
Error deploying Strimzi-Operator from Azure Repo

### DISCUSSION THREAD

**Discussion by atuss87** [strimzi/strimzi-kafka-operator#8989] [Q&A]
I am experiencing trouble deploying a Strimzi-Operator using either an IronBank or a public image when pushed to an Azure repository. I am only changing the respective image location values (either public image or the IronBank/public image located in my Azure repo) within the '060-Deployment-strimzi-cluster-operator.yaml' file. When I point to the public repo, all is well. The pod deploys and runs without error, however, when I try to deploy the pod from the same image (or IronBank image) located in my Azure repo, I get the following error: 

Normal   Created    53m (x5 over 55m)    kubelet            Created container strimzi-cluster-operator
Warning  Failed     53m (x5 over 55m)    kubelet            Error: failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/opt/strimzi/bin/cluster_operator_run.sh": stat /opt/strimzi/bin/cluster_operator_run.sh: no such file or directory: unknown
Warning  BackOff    22s (x276 over 55m)  kubelet            Back-off restarting failed container

The redacted version of my 060-Deployment-strimzi-cluster-operator.yaml file looks like this:

apiVersion: apps/v1
kind: Deployment
metadata:
  name: strimzi-cluster-operator
  labels:
    app: strimzi
spec:
  replicas: 1
  selector:
    matchLabels:
      name: strimzi-cluster-operator
      strimzi.io/kind: cluster-operator
  template:
    metadata:
      labels:
        name: strimzi-cluster-operator
        strimzi.io/kind: cluster-operator
    spec:
      imagePullSecrets:
        - name: REDACTED
      serviceAccountName: strimzi-cluster-operator
      volumes:
        - name: strimzi-tmp
          emptyDir:
            medium: Memory
            sizeLimit: 1Mi
        - name: co-config-volume
          configMap:
            name: strimzi-cluster-operator
      containers:
        - name: strimzi-cluster-operator
          image: REDACTED/strimzi/operator:0.36.0
          ports:
            - containerPort: 8080
              name: http
          args:
            - /opt/strimzi/bin/cluster_operator_run.sh
          volumeMounts:
            - name: strimzi-tmp
              mountPath: /tmp
            - name: co-config-volume
              mountPath: /opt/strimzi/custom-config/
          env:
            - name: STRIMZI_IMAGE_PULL_SECRETS
              value: REDACTED
            - name: STRIMZI_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: STRIMZI_FULL_RECONCILIATION_INTERVAL_MS
              value: "120000"
            - name: STRIMZI_OPERATION_TIMEOUT_MS
              value: "300000"
            - name: STRIMZI_DEFAULT_TLS_SIDECAR_ENTITY_OPERATOR_IMAGE
              value: REDACTED/strimzi/kafka:0.36.0-kafka-3.5.0
            - name: STRIMZI_DEFAULT_KAFKA_EXPORTER_IMAGE
              value: REDACTED/strimzi/kafka:0.36.0-kafka-3.5.0
            - name: STRIMZI_DEFAULT_CRUISE_CONTROL_IMAGE
              value: REDACTED/strimzi/kafka:0.36.0-kafka-3.5.0
            - name: STRIMZI_KAFKA_IMAGES
              value: |
                3.4.0=quay.io/strimzi/kafka:0.36.0-kafka-3.4.0
                3.4.1=quay.io/strimzi/kafka:0.36.0-kafka-3.4.1
                3.5.0=REDACTED/strimzi/kafka:0.36.0-kafka-3.5.0
            - name: STRIMZI_KAFKA_CONNECT_IMAGES
              value: |
                3.4.0=quay.io/strimzi/kafka:0.36.0-kafka-3.4.0
                3.4.1=quay.io/strimzi/kafka:0.36.0-kafka-3.4.1
                3.5.0=REDACTED/strimzi/kafka:0.36.0-kafka-3.5.0
            - name: STRIMZI_KAFKA_MIRROR_MAKER_IMAGES
              value: |
                3.4.0=quay.io/strimzi/kafka:0.36.0-kafka-3.4.0
                3.4.1=quay.io/strimzi/kafka:0.36.0-kafka-3.4.1
                3.5.0=REDACTED/strimzi/kafka:0.36.0-kafka-3.5.0
            - name: STRIMZI_KAFKA_MIRROR_MAKER_2_IMAGES
              value: |
                3.4.0=quay.io/strimzi/kafka:0.36.0-kafka-3.4.0
                3.4.1=quay.io/strimzi/kafka:0.36.0-kafka-3.4.1
                3.5.0=REDACTED/strimzi/kafka:0.36.0-kafka-3.5.0
            - name: STRIMZI_DEFAULT_TOPIC_OPERATOR_IMAGE
              value: REDACTED/strimzi/operator:0.36.0
            - name: STRIMZI_DEFAULT_USER_OPERATOR_IMAGE
              value: REDACTED/strimzi/operator:0.36.0
            - name: STRIMZI_DEFAULT_KAFKA_INIT_IMAGE
              value: REDACTED/strimzi/operator:0.36.0
            - name: STRIMZI_DEFAULT_KAFKA_BRIDGE_IMAGE
              value: REDACTED/kafka-bridge:0.26.0
            - name: STRIMZI_DEFAULT_KANIKO_EXECUTOR_IMAGE
              value: REDACTED/kaniko-executor:0.36.0
            - name: STRIMZI_DEFAULT_MAVEN_BUILDER
              value: REDACTED/maven-builder:0.36.0
            - name: STRIMZI_OPERATOR_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: STRIMZI_FEATURE_GATES
              value: ""
            - name: STRIMZI_LEADER_ELECTION_ENABLED
              value: "true"
            - name: STRIMZI_LEADER_ELECTION_LEASE_NAME
              value: "strimzi-cluster-operator"
            - name: STRIMZI_LEADER_ELECTION_LEASE_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: STRIMZI_LEADER_ELECTION_IDENTITY
              valueFrom:
                fieldRef:
                  fieldPath: metadata.name
          livenessProbe:
            httpGet:
              path: /healthy
              port: http
            initialDelaySeconds: 10
            periodSeconds: 30
          readinessProbe:
            httpGet:
              path: /ready
              port: http
            initialDelaySeconds: 10
            periodSeconds: 30
          resources:
            limits:
              cpu: 1000m
              memory: 384Mi
            requests:
              cpu: 200m
              memory: 384Mi

Does anyone know how to remediate or resolve this error? When I change the above redacted values to the public image (quay.io/strimzi/kafka:0.36.0-kafka-3.5.0 and quay.io/strimzi/operator:0.36.0, respectively) it deploys and runs without error. Thanks in advance for any help!

**Comment by scholzj** [comment]
How do you copy the images? My guess is that you try to do something simple such as:
* Pull the image on your local computer with `docker pull`
* Tag it to the new repo with `docker tag`
* Push it to your other repo with `docker push`

If that is what you do and you do it for example from an Arm based Mac, it would copy the Arm64 images which would not work if your servers are Amd64. That would create an error like this. So you should either make sure you copy the right platform or use some tool such as `regctl` which would copy the whole manifest.

**Reply by atuss87** [reply]
Yes, that's exactly how I copy the images. I'll look into regctl, thanks for the respone.
